src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -1069,49 +995,39 @@ def run_datamover(self, test_desc=None,
 
         return result
 
-    def run_dm_activities_with_ior(self, tool, create_dataset=False, pool=None, cont=None):
-        """Generic method to perform varios datamover activities
+    def run_dm_activities_with_ior(self, tool, pool, cont, create_dataset=False):
+        """Generic method to perform various datamover activities
            using ior
         Args:
-            tool(str): specify the tool name to be used
-            create_dataset(bool): boolean to create initial set of
-                                  data using ior. Defaults to False.
-            pool(TestPool): Pool object. Defaults to None
-            cont(TestContainer): Container object. Defaults to None
+            tool (str): specify the tool name to be used
+            pool (TestPool): source pool object
+            cont (TestContainer): source container object
+            create_dataset (bool): boolean to create initial set of
+                                   data using ior. Defaults to False.
         """
         # Set the tool to use
         self.set_tool(tool)
 
         if create_dataset:
-            # create initial datasets
-            if not pool:
-                pool = self.create_pool()
-            cont = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
-
-            # update and run ior on container 1
+            # create initial datasets with ior
             self.run_ior_with_params("DAOS", self.ior_cmd.test_file.value, pool, cont)
-        else:
-            if not pool:
-                pool = self.pool[0]
-            if not cont:
-                cont = self.container[-1]
 
         # create cont2
-        cont2 = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
+        cont2 = self.get_container(pool, oclass=self.ior_cmd.dfs_oclass.value)
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'oclass' in method call